lib/api.ts

```diff
@@ -44,18 +44,18 @@ export const getAllCategoryPostPaths = (category: CategoryOption): string[] => {
 export const getSingleCategoryPost = (
   slug: string[] | string,
   catPath: string,
-): GrayMatterFile<string> | void => {
+): { [k: string]: any } | void => {
   const postPath = typeof slug === 'string' ? slug : slug.join('/');
   const filePath = path.join(catPath, `${postPath}.md`);
   const indexPath = path.join(catPath, `${postPath}/index.md`);
 
   // If file exists, it is not an index page.
   if (fs.existsSync(filePath)) {
-    return matter(fs.readFileSync(filePath, 'utf8'));
+    return { isIndex: false, ...matter(fs.readFileSync(filePath, 'utf8')) };
   }
 
   if (fs.existsSync(indexPath)) {
-    return matter(fs.readFileSync(indexPath, 'utf8'));
+    return { isIndex: true, ...matter(fs.readFileSync(indexPath, 'utf8')) };
```

Comment:
Added this boolean so we can use it to differentiate single post vs. index page layouts